weblab/datasets/tests/test_models.py

```diff
@@ -37,6 +66,19 @@ def test_visibility_and_sharing(self, logged_in_user, other_user, anon_user):
 
         # TODO - No testing of shared datasets - waiting for implementation in front end
 
+    def test_column_names(self, dataset_creator):
```

Comment:
It would be worth adding a test with a tab-separated file too, since some of our datasets are (but still called .csv). Perhaps also a test like `b'"column 1","column 2"'`.

And one with two files where the second is the main file?